
<launch>
    <!-- number of moving objects -->
    <arg name="obj_num" value="10" />
    <arg name="drone_id" value="0"/>

    <arg name="map_size_x" value="20.0"/>
    <arg name="map_size_y" value="10.0"/>
    <arg name="map_size_z" value=" 4.0"/>
    <arg name="odom_topic" value="visual_slam/odom" />
    
    <!-- map -->
    <node pkg ="dynamic_map_generator" name ="dynamic_map_generator" type ="dynamic_map_generator_node" output = "screen">
        <param name="sensing/rate"   value="10.0"/>   
        <param name="map/x_size"     value="$(arg map_size_x)" />
        <param name="map/y_size"     value="$(arg map_size_y)" />
        <param name="map/z_size"     value="$(arg map_size_z)" />
        <param name="map/resolution" value="0.1"/>        
        <param name="map/frame_id"     value="map" />
        <param name="map/seed" value="1"/>
        <param name="map/obs_num"    value="20"/>
    
        <param name="obstacle/upper_vel" value="1.0"/>

        <param name="aabb/lower_x" value="1.0"/>
        <param name="aabb/upper_x" value="2.0"/>
        <param name="aabb/lower_y" value="1.0"/>
        <param name="aabb/upper_y" value="2.0"/>        
        <param name="aabb/lower_z" value="2.0"/>
        <param name="aabb/upper_z" value="5.0"/>        

        <param name="cylinder/lower_rad" value="0.2"/>
        <param name="cylinder/upper_rad" value="0.5"/>
        <param name="cylinder/lower_hei" value="4.0"/>
        <param name="cylinder/upper_hei" value="4.0"/>        
    
        <param name="circlegate/radius_l" value="0.7"/>        
        <param name="circlegate/radius_h" value="2.5"/>        
        <param name="circlegate/thickness" value="0.1"/>        
        <param name="circlegate/dr" value="0.3"/>        
        <param name="circlegate/theta" value="1.2"/>        
        <param name="enable_ground" value="true"/>
    </node>

    <!-- use simulator -->
    <!-- <include file="$(find ego_planner)/launch/simulator.xml">
        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <arg name="init_x_" value="-15"/>
        <arg name="init_y_" value="0"/>
        <arg name="init_z_" value="0.1"/>
        <arg name="odometry_topic" value="$(arg odom_topic)" />
    </include> -->

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dynamic_map_generator)/config/default.rviz" required="true" />
      
</launch>
